### ISSUE TITLE
Feature Request: Record Timer

### ISSUE THREAD

**Issue by ArchUser** [chrippa/livestreamer#812]
Not sure how much work this would take but would it be possible to have a record time? Like if I can tell livestreamer to stop saving to file after 2 hours. There is already a timer for how long the video has been recording for, it doesn't seem like it should be that hard to have it kill the program after x time.

**Comment by dirk362** [comment]
There's a posting in the google forum for livestreamer with a 'patch' that provides a timer. 
See https://groups.google.com/forum/#!topic/livestreamer/BMlxr9wQCkc

It would be very useful to have this capability within the core/main release as other forks are behind the main code base. 

Not suggesting necessarily that the patch is appropriate to just go into the main code base as is, but might be a useful starting point.

Appreciate all your hard work. Keep it up :)

**Comment by j4ybeee** [comment]
Yes, it would be great to have an option to allow Livestreamer to record/capture a file for X time. The fork that was created looks like a simple solution.

**Comment by 0880** [comment]
>I tried looking for a batch script option for this but there doesn't seem to be a very efficient command.

Batch file:

````
start livestreamer.exe "hls://nasatv-lh.akamaihd.net/i/NASA_101@319270/index_1000_av-p.m3u8" best -o 01.ts
timeout /t 1800
taskkill /im livestreamer.exe /f
````

**Comment by j4ybeee** [comment]
I'm on Win 7 and currently use the Windows Task Scheduler to start the batch job and also to end it after x minutes/hours. It works well for me, although as I say having an internal Livestreamer record timer would still be useful.
